addons/account/models/account_move.py

```diff
@@ -2643,8 +2643,16 @@ def _get_quick_edit_suggestions(self):
                     self.journal_id.company_id.account_purchase_tax_id
                 )
             taxes = self.fiscal_position_id.map_tax(taxes)
-        price_untaxed = taxes.with_context(force_price_include=True).compute_all(
-            self.quick_edit_total_amount - self.tax_totals['amount_total'])['total_excluded']
+
+        # When a payment term has an early discount set to "Always (upon invoice)", recomputing the untaxed amount
+        # should take in consideration the discount percentage otherwise we'd get a wrong value.
+        if self.invoice_payment_term_id.early_pay_discount_computation == 'mixed':
+            discount_percentage = self.invoice_payment_term_id.discount_percentage
+            price_untaxed = taxes.with_context(force_price_include=True, early_pay_discount=True, discount_percentage=discount_percentage).compute_all(
+                self.quick_edit_total_amount - self.tax_totals['amount_total'])['total_excluded']
+        else:
+            price_untaxed = taxes.with_context(force_price_include=True).compute_all(
+                self.quick_edit_total_amount - self.tax_totals['amount_total'])['total_excluded']
```

Comment:
Let's clarify something. We you have a base of 100, 10% tax avec 5% of early payment discount, you expect a tax amount of 95 x 0.1 = 9.5 instead of 10.
So here,
```py
discount_percentage == 5
self.quick_edit_total_amount == 109.5
self.tax_totals['amount_total'] == 0.0
```
Mathematically, there is absolutely no way to retrieve 100 & 9.5 exactly. If you exclude fixed taxes and make it working only for a single tax, you could try another approach.
x + ((x * 0.95) * 0.1) = y
x * 0.95 * 0.1 = y - x
0.95 * 0.1 = (y - x) / x
0.95 * 0.1 = y/x - 1
(0.95 * 0.1) + 1 = y / x
x = y / ((0.95 * 0.1) + 1)
So
x = <quick_edit_total_amount> / (((100 - <discount_percentage>) * <tax_%>) + 1)
x = 109.5 / ((95 * 0.1) + 1) = 100

However, this works only for percentage tax :/